### ISSUE TITLE
Consolidate ZoneEquipmentManager::CalcZoneLeavingConditions and RoomAirModelUserTempPattern::SetSurfHBDataForTempDistModel

### ISSUE THREAD

**Issue by mjwitte** [NREL/EnergyPlus#6322]
Issue overview
--------------
After #6272 drops in (and likely after v8.8 release) consolidate `ZoneEquipmentManager::CalcZoneLeavingConditions` and `RoomAirModelUserTempPattern::SetSurfHBDataForTempDistModel` to use a common function for calculating the return node temp(s) for a given zone.  Currently RoomAirModelUserTempPattern::SetSurfHBDataForTempDistModel has a [block of code taken originally from CalcZoneLeavingConditions](https://github.com/NREL/EnergyPlus/blob/6489e202ae11db706dae1ea4d6b7589266d52302/src/EnergyPlus/RoomAirModelUserTempPattern.cc#L1067-L1179). For #6272 appropriate changes were made in both places to allow for multiple return nodes. These two functions should ultimately share a common function, but that's better done as a separate pull request for clarity.

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version)
 - Version of EnergyPlus (if using an intermediate build, include SHA)
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)
